README.md

```diff
@@ -1,74 +1,372 @@
+that data according to the [Message Encryption for Web Push spec](https://tools.ietf.org/html/draft-ietf-webpush-encryption).
 
-## sendNotification(endpoint, params)
+This module makes it easy to send messages and will also handle legacy support
+for browsers relying on GCM for message sending / delivery.
 
-Send a Push notification to an endpoint. *params* contains optional parameters:
-- *TTL* is a value in seconds that describes how long a push message is retained by the push service (by default, four weeks);
-- *userPublicKey* is the public key of the receiver (from the browser);
-- *userAuth* is the auth secret of the receiver (from the browser);
-- *payload* is the message to attach to the notification.
-- *vapid* an object with parameters for [VAPID](https://tools.ietf.org/html/draft-thomson-webpush-vapid-02).
+# Install
 
-Note that, in order to encrypt the *payload*, *userPublicKey* and *userAuth* are required.
+Installation is a simple, just install via npm.
 
-The properties of the *vapid* objects are:
-- *subject*, a contact URI for the application server (either 'mailto:' or 'https:');
-- *privateKey*;
-- *publicKey*.
+    npm install web-push --save
+
+# Usage
+
+The common use case for this library is an application server using
+a GCM API key and VAPID keys.
+
+    const webpush = require('web-push');
+
+    // VAPID keys should only be generated only once.
+    const vapidKeys = webpush.generateVAPIDKeys();
+
+    webpush.setGCMAPIKey('<Your GCM API Key Here>');
+    webpush.setVapidDetails(
+      'mailto:example@yourdomain.org',
+      vapidKeys.publicKey,
+      vapidKeys.privateKey
+    );
+
+    // This is the same output of calling JSON.stringify on a PushSubscription
+    const pushSubscription = {
+      endpoint: '.....',
+      keys: {
+        auth: '.....',
+        p256dh: '.....'
+      }
+    };
+
+    webpush.sendNotification(pushSubscription, 'Your Push Payload Text');
+
+## Using VAPID Key for applicationServerKey
+
+When using your VAPID key in your web app, you'll need to convert the
+URL safe base64 string in a Uint8Array to pass into the subscribe call,
+which you can do like so:
+
+    function urlBase64ToUint8Array(base64String) {
+      const padding = '='.repeat((4 - base64String.length % 4) % 4);
+      const base64 = (base64String + padding)
+        .replace(/\-/g, '+')
+        .replace(/_/g, '/');
+
+      const rawData = window.atob(base64);
+      const outputArray = new Uint8Array(rawData.length);
+
+      for (let i = 0; i < rawData.length; ++i) {
+        outputArray[i] = rawData.charCodeAt(i);
+      }
+      return outputArray;
+    }
+
+    const vapidPublicKey = '<Your Public Key from generateVAPIDKeys()>';
+    const convertedVapidKey = urlBase64ToUint8Array(vapidPublicKey);
+
+    registration.pushManager.subscribe({
+      userVisibleOnly: true,
+      applicationServerKey: convertedVapidKey  
+    });
+
+# API Referance
+
+## sendNotification(pushSubscription, payload, options)
```

Comment:
Can we somehow specify that `sendNotification` can also be used without sending a payload and/or without using VAPID?